app/code/Magento/ImportExport/Controller/Adminhtml/Import/Download.php

```diff
@@ -68,22 +78,25 @@ public function __construct(
      */
     public function execute()
     {
-        $fileName = $this->getRequest()->getParam('filename') . '.csv';
-        $moduleDir = $this->componentRegistrar->getPath(ComponentRegistrar::MODULE, self::SAMPLE_FILES_MODULE);
-        $fileAbsolutePath = $moduleDir . '/Files/Sample/' . $fileName;
-        $directoryRead = $this->readFactory->create($moduleDir);
-        $filePath = $directoryRead->getRelativePath($fileAbsolutePath);
+        $entityName = $this->getRequest()->getParam('filename');
+
+        if (preg_match('/^\w+$/', $entityName) == 0) {
+            /** @var \Magento\Backend\Model\View\Result\Redirect $resultRedirect */
+            $this->messageManager->addErrorMessage(__('Incorrect file name.'));
```

Comment:
Wrong error message
But in this case, we have incorrect `$entityName`(we resolve path by `$entityName`) but not file name

Reply:
Done.